sdks/java/container/Dockerfile

```diff
@@ -15,8 +15,9 @@
 #  See the License for the specific language governing permissions and
 # limitations under the License.
 ###############################################################################
+ARG base_image
 ARG java_version
-FROM eclipse-temurin:${java_version}
+FROM ${base_image}:${java_version}
```

Comment:
we switched from openjdk to temurin in #25402 due to that openjdk image for java8 and 11 was deprecated. For now we still use temurin for 8,11,17; openjdk for 21 temporarily until upstream https://adoptium.net/blog/2023/09/temurin21-delay/ resolved